cmd/buildah/push.go

```diff
@@ -175,6 +175,7 @@ func pushCmd(c *cobra.Command, args []string, iopts pushResults) error {
 	} else {
 		logrus.Debugf("pushed image with digest %s", digest.String())
 	}
+	logrus.Debugf("Successfully pushed %s@%s\n", dest.StringWithinTransport(), digest.String())
```

Comment:
If this does not have to restore the previous string exactly, I’d prefer
```suggestion
	logrus.Debugf("Successfully pushed to %s with digest %s\n", transports.ImageName(dest), digest.String())
```
e.g. `Successfully pushed to docker://example.com/busybox:latest with digest sha256:aaaa…aaaa`; that way the “pushed to…” string is actually a valid reference, unlike the use of either `StringWithinTransport` or joining the result of that with `@`_digest_.

Ultimately, as long as it is a debug text and not an API promise (or maybe it _is_ intended to be a promise per #1744 ??), anything goes, of course.

Reply:
I like the idea of --verbose and then output the information that @mtrmac suggests, as well as logging it as a debugf.

Reply:
OTOH see https://github.com/containers/buildah/pull/1744#issuecomment-522233304 , do we want to make an actual safe API (instead of / as well as) strings in a stream of text?

(The API concern is not a reason not to add this text as well.)

Reply:
OK, I've taken @mtrmac suggestion for the "Successfully pushed ..." line and kept it as a Debug statement.  As noted in #1744 I've created a --digestfile option and now write out the digest which is the value found in `digest.String()` and that looks something like:  `sha256:b552b401d9977f4da64e410727e84a736be5b78f96b561a9b21579b3d1d74a8f`.  That's all the stuff to the right of the `@` previously.   @ wwilson  if that needs to be adjusted for your use case, please let us know.

I'll be pushing a new commit to #1744 after a bit more testing.

Reply:
oops, to many prs, I'll repush here at #1799.